src/main/java/net/dv8tion/jda/core/entities/MessageHistory.java

```diff
@@ -330,4 +332,164 @@ public Message getMessageById(long id)
+     * of messages sent before the mentioned message ID (exclusive).
+     * <br>The provided id need not be valid!
+     *
+     * @param  channel
+     *         The {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}
+     * @param  messageId
+     *         The pivot ID to use
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If any of the provided arguments is {@code null}
+     * @throws net.dv8tion.jda.core.exceptions.InsufficientPermissionException
+     *         If this is a TextChannel and the currently logged in account does not
+     *         have the permission {@link net.dv8tion.jda.core.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}
+     *
+     * @return {@link net.dv8tion.jda.core.entities.MessageHistory.MessageRetrieveAction MessageRetrieveAction}
+     */
+    public static MessageRetrieveAction getHistoryBefore(MessageChannel channel, String messageId)
+    {
+        Checks.noWhitespace(messageId, "Message ID");
+        Checks.notNull(channel, "Channel");
+        if (channel.getType() == ChannelType.TEXT )
+        {
+            TextChannel t = (TextChannel) channel;
+            if (!t.getGuild().getSelfMember().hasPermission(t, Permission.MESSAGE_HISTORY, Permission.MESSAGE_READ))
+                throw new InsufficientPermissionException(Permission.MESSAGE_HISTORY);
+        }
+        Route.CompiledRoute route = Route.Messages.GET_MESSAGE_HISTORY.compile(channel.getId()).withQueryParams("before", messageId);
+        return new MessageRetrieveAction(route, channel);
+    }
+
+    /**
+     * Constructs a {@link net.dv8tion.jda.core.entities.MessageHistory MessageHistory} with the initially retrieved history
+     * of messages sent around the mentioned message ID (inclusive).
+     * <br>The provided id need not be valid!
+     *
+     * @param  channel
+     *         The {@link net.dv8tion.jda.core.entities.MessageChannel MessageChannel}
+     * @param  messageId
+     *         The pivot ID to use
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If any of the provided arguments is {@code null}
+     * @throws net.dv8tion.jda.core.exceptions.InsufficientPermissionException
+     *         If this is a TextChannel and the currently logged in account does not
+     *         have the permission {@link net.dv8tion.jda.core.Permission#MESSAGE_HISTORY Permission.MESSAGE_HISTORY}
+     *
+     * @return {@link net.dv8tion.jda.core.entities.MessageHistory.MessageRetrieveAction MessageRetrieveAction}
+     */
+    public static MessageRetrieveAction getHistoryAround(MessageChannel channel, String messageId)
+    {
+        Checks.noWhitespace(messageId, "Message ID");
+        Checks.notNull(channel, "Channel");
+        if (channel.getType() == ChannelType.TEXT )
+        {
+            TextChannel t = (TextChannel) channel;
+            if (!t.getGuild().getSelfMember().hasPermission(t, Permission.MESSAGE_HISTORY, Permission.MESSAGE_READ))
```

Comment:
No need to check for Permission.MESSAGE_READ explicitly anymore, consider removing it.